source/wp-content/themes/wporg-parent-2021/sass/base/_elements.scss

```diff
@@ -1,3 +1,23 @@
+// Create balanced text blocks
+h1,
+h2,
+h3,
+h4,
+h5,
+h6,
+caption,
+figcaption {
+	text-wrap: var(--wp--custom--heading--typography--text-wrap);
+}
+
+// Prevent orphans
+p,
```

Comment:
Do you think this will [cause performance issues](https://developer.chrome.com/blog/css-text-wrap-balance/#limitations) since we're targeting every `<p>` tag?

Reply:
Thinking about the the [extra p tag](WordPress/wporg-developer/pull/430) problem on developer.

Reply:
Yeah good point. Those limitations apply to `text-wrap: balance` which we're not using on p tags, only headings as [recommended](https://developer.chrome.com/blog/css-text-wrap-balance/#performance_considerations).

The [post](https://developer.chrome.com/blog/css-text-wrap-pretty) on `text-wrap: pretty` doesn't mention any limitations.

Reply:
Saw [someone](https://twitter.com/csswizardry/status/1725543814587134435) has tested the performance of text-wrap: pretty; and text-wrap: balance; and found that thier overhead are quite similar, with balance being even a bit faster. Might be due to its limitation of only working for six wrapped lines and under. 
Overall the performance overhead for both is negligible.

And also saw that - applying balance to large paragraphs will have no effect and still comes at a performance cost because the browser is trying to calculate optimal balance even though it won't apply anything.